pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -324,11 +285,21 @@ func getLoginServer(registry containerregistry.Registry) string {
 }
 
 func getACRDockerEntryFromARMToken(a *acrProvider, loginServer string) (*credentialprovider.DockerConfigEntry, error) {
-	// Run EnsureFresh to make sure the token is valid and does not expire
-	if err := a.servicePrincipalToken.EnsureFresh(); err != nil {
-		klog.Errorf("Failed to ensure fresh service principal token: %v", err)
-		return nil, err
+	if a.servicePrincipalToken == nil {
+		token, err := auth.GetServicePrincipalToken(a.config, a.environment)
+		if err != nil {
+			klog.Errorf("Failed to create service principal token: %v", err)
```

Comment:
Isn't the log message has to be like `Failed to get service principal token: %v", err`

Reply:
aha, good catch! let me work out another typo fix.